### PR TITLE
NEXUS-8821 - improve logging for failed auth token request

### PR DESCRIPTION
**Pull request by peterlynch**
make it easier to diagnose username mismatch errors

https://issues.sonatype.org/browse/NEXUS-8821

### CODE REVIEW

File: plugins/internal/nexus-wonderland-plugin/src/main/java/org/sonatype/nexus/wonderland/rest/AuthenticateResource.java [hunk 1]
```diff
@@ -87,13 +87,17 @@ public AuthenticateResource(final SecuritySystem security,
...
+    final String password = Tokens.decodeBase64String(token.getP());
 
     // Require current user to be the requested user to authenticate
     Subject subject = security.getSubject();
-    if (!subject.getPrincipal().toString().equals(username)) {
+    final String principalName =  subject.getPrincipal().toString();
+    log.debug("payload username: {}, payload password: {}, principal: {}", username, Tokens.mask(password), principalName);
+
+    if (!principalName.equals(username)) {
+      log.warn("access token request denied - authenticated user {} does not match payload user {}",
```
Comment (jdillon) [inline plugins/internal/nexus-wonderland-plugin/src/main/java/org/sonatype/nexus/wonderland/rest/AuthenticateResource.java]: auth-token not access token probably clearer to devs.

**Comment by jdillon** [comment]
+1 after the string is updated

**Comment by cstamas** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by jdillon** [comment]
should update master as well, it still has this endpoint so we should update it as well.

**Comment by peterlynch** [comment]
I will make a master PR when Rich is back

**Pull request merged**